### PR TITLE
[R20-851] switching to new menu endpoint

### PR DESCRIPTION
**Pull request by lambry**
**Issue**: https://digital-vic.atlassian.net/browse/R20-851

### What I did

- Switched over to the new menu endpoint, baseUrl needs to be set to: https://nginx-php.pr-203.content-reference-sdp-vic-gov-au.sdp4.sdp.vic.gov.au

Screenshot 2023-04-14 at 12 32 24 pm

### CODE REVIEW

**Review by jeffdowdle — CHANGES_REQUESTED** [review]
This is looking really good, my main question is, do you think we need to support both of these with a feature flag at this stage because not all backends will have the new endpoint?

**Comment by lambry** [comment]
> This is looking really good, my main question is, do you think we need to support both of these with a feature flag at this stage because not all backends will have the new endpoint?

@jeffdowdle Good question, the backend update for this is actually already in a tide release https://github.com/dpc-sdp/tide_api/releases/tag/3.0.10 so it's ahead of the curve there.

**Comment by lambry** [comment]
> This is looking really good, my main question is, do you think we need to support both of these with a feature flag at this stage because not all backends will have the new endpoint?

Hey @jeffdowdle a 'flag' now exists so the 'single' or 'paginated' endpoint can be used. Here's a screenshot for each running, notice the menu or is different.

## New (Single request)
single

## Old (Paginated request)
paginated

**Review by dylankelly — APPROVED** [review]

File: packages/ripple-tide-api/src/services/lib/site-menu.ts [hunk 1]
```diff
@@ -87,17 +87,18 @@ const _setActivePath = function (branch, path) {
 const getHierarchicalMenu = function (menu, activeUrl) {
   const linkValues = [] as any
   for (const link of menu) {
-    if (link.attributes.enabled) {
```
Comment (jeffdowdle) [inline packages/ripple-tide-api/src/services/lib/site-menu.ts:90]: We should bring back to check, something like 
`if (!IsOldMenuEndpoint || (IsOldMenuEndpoint && link.attributes.enabled) {`

Comment (lambry) [reply packages/ripple-tide-api/src/services/lib/site-menu.ts:90]: Hey, this is done. I've just called it `enabledCheck`.

**Review by jeffdowdle — APPROVED** [review]

File: packages/ripple-tide-api/src/services/tide-api-base.ts [hunk 2]
```diff
@@ -74,22 +76,54 @@ export default class TideApiBase extends HttpClient {
...
 
-  async getSiteMenu(siteId, menuData, activePath?) {
+  async getSiteMenu(siteId, menuData, activePath = null) {
     if (!menuData) {
       return null
     }
 
     const menuName = menuData.drupal_internal__id
 
+    if (this.tide?.menuEndpoint === 'paginated') {
```
[the remark this answers is not in the document]
Comment (lambry) [reply packages/ripple-tide-api/src/services/tide-api-base.ts:88]: @jeffdowdle Good idea, that's done now.

**Pull request merged**